sdks/python/apache_beam/transforms/external.py

```diff
@@ -319,25 +319,9 @@ def _service(self):
     if isinstance(self._expansion_service, str):
       channel_options = [("grpc.max_receive_message_length", -1),
                          ("grpc.max_send_message_length", -1)]
-      if hasattr(grpc, 'local_channel_credentials'):
-        # Some environments may not support insecure channels. Hence use a
-        # secure channel with local credentials here.
```

Comment:
My concern is that we are breaking those environments by removing the code here. I'm not a GRPC expert but if we are convinced that insecure channels can always be used, then I'm ok with the change. Otherwise we probably need some kind of fallback code which attempts a secure channel first, then falls back to using an insecure channel.

Reply:
@nirga - Do you know the answer? I think the suggestion to use fallback mechanism is a good one.

Reply:
@mxm I think the issue here is not whether the channel is secure or insecure. The point is `local_channel_credentials` only allows local connections so we can't use the expansion service running on a remote server. This PR makes sense to me.

Reply:
@mxm maybe we can allow people to explicitly set this, instead of automatically switching to local channels?

Reply:
There are two problems here: 

1. The issue the MR is trying to solve, i.e. allowing for remote connections 
2. The problem of insecure channels not being supported in all environments (possible regression caused by this MR)

When fixing a bug we must ensure not to add another bug. This is my only concern.

Reply:
(2) is also confirmed here: https://github.com/apache/beam/pull/14671#pullrequestreview-741061287

Reply:
On this note, we should probably have a test to prevent (2) from breaking.